Date: Mon, 16 Feb 2026 12:12:06 +0100
Subject: [PATCH 2/2] x1600: add support for early printk on UART2

earlycon is broken for some reason; this is the only possibility of getting reliable early boot logs on this platform 
---
 arch/mips/Kconfig        | 3 +++
 arch/mips/kernel/setup.c | 6 ++++++
 2 files changed, 9 insertions(+)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 83445a2cb..91bf19333 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -427,6 +427,9 @@ config MACH_INGENIC_SOC
 	select SYS_SUPPORTS_ZBOOT_UART16550
 	select CPU_SUPPORTS_CPUFREQ
 	select MIPS_EXTERNAL_TIMER
+	select SYS_HAS_EARLY_PRINTK
+	select USE_GENERIC_EARLY_PRINTK_8250
+
 
 config LANTIQ
 	bool "Lantiq based platforms"
diff --git a/arch/mips/kernel/setup.c b/arch/mips/kernel/setup.c
index 12a1a4ffb..596578a7e 100644
--- a/arch/mips/kernel/setup.c
+++ b/arch/mips/kernel/setup.c
@@ -774,6 +774,12 @@ void __init setup_arch(char **cmdline_p)
 {
 	cpu_probe();
 	mips_cm_probe();
+
+	if (IS_ENABLED(CONFIG_EARLY_PRINTK_8250) &&
+	    IS_ENABLED(CONFIG_MACH_INGENIC_SOC) &&
+	    IS_ENABLED(CONFIG_MACH_X1600))
+		setup_8250_early_printk_port(CKSEG1ADDR(0x10032000), 2, 0);
+
 	prom_init();
 
 	setup_early_fdc_console();
-- 
2.52.0

